### PR TITLE
bugfix/12892: localize plugin runtime error notifications

### LINKED ISSUE #12892: Plugin runtime error notification is not localized
### Bug Description:
Plugin runtime error notification is not localized 

### Steps to Reproduce:

image

https://github.com/eclipse-theia/theia/blob/cb426569d1d5fe42567f5ec5d35b3c77a92f3295/packages/plugin-ext/src/hosted/node/hosted-plugin-process.ts#L220-L224

### Additional Information

- Operating System: any
- Theia Version: the latest

### PR DESCRIPTION
**Pull request by Zebsterpasha**
#### What it does
Fixes #12892 

#### How to test
Just check the localization keys

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/plugin-ext/src/hosted/node/hosted-plugin-process.ts [hunk 1]
```diff
@@ -217,11 +217,11 @@ export class HostedPluginProcess implements ServerPluginRunner {
...
         this.logger.error(`[${serverName}: ${pid}] IPC exited, with signal: ${signal}, and exit code: ${code}`);
 
-        const message = 'Plugin runtime crashed unexpectedly, all plugins are not working, please reload the page.';
-        let hintMessage: string = 'If it doesn\'t help, please check Theia server logs.';
+        const message = nls.localize('theia/plugin-ext/runtimeErrorMessage', 'Plugin runtime crashed unexpectedly, all plugins are not working, please reload the page.');
+        let hintMessage: string = nls.localize('theia/plugin-ext/runtimeErrorHintMessage', 'If it doesn\'t help, please check Theia server logs.');
         if (signal && signal.toUpperCase() === 'SIGKILL') {
             // May happen in case of OOM or manual force stop.
-            hintMessage = 'Probably there is not enough memory for the plugins. ' + hintMessage;
+            hintMessage = nls.localize('theia/plugin-ext/runtimeErrorAddHingMessage', 'Probably there is not enough memory for the plugins. ') + hintMessage;
```
Comment (vince-fugnitto) [inline packages/plugin-ext/src/hosted/node/hosted-plugin-process.ts:224]: minor - typo:

```suggestion
            hintMessage = nls.localize('theia/plugin-ext/runtimeErrorAddingMessage', 'Probably there is not enough memory for the plugins. ') + hintMessage;
```

---

I believe we can simplify to `runtimeMemoryError`,

File: packages/plugin-ext/src/hosted/node/hosted-plugin-process.ts [hunk 2]
```diff
@@ -217,11 +217,11 @@ export class HostedPluginProcess implements ServerPluginRunner {
...
         this.logger.error(`[${serverName}: ${pid}] IPC exited, with signal: ${signal}, and exit code: ${code}`);
 
-        const message = 'Plugin runtime crashed unexpectedly, all plugins are not working, please reload the page.';
-        let hintMessage: string = 'If it doesn\'t help, please check Theia server logs.';
+        const message = nls.localize('theia/plugin-ext/runtimeErrorMessage', 'Plugin runtime crashed unexpectedly, all plugins are not working, please reload the page.');
+        let hintMessage: string = nls.localize('theia/plugin-ext/runtimeErrorHintMessage', 'If it doesn\'t help, please check Theia server logs.');
         if (signal && signal.toUpperCase() === 'SIGKILL') {
             // May happen in case of OOM or manual force stop.
-            hintMessage = 'Probably there is not enough memory for the plugins. ' + hintMessage;
+            hintMessage = nls.localize('theia/plugin-ext/runtimeMemoryError', 'Probably there is not enough memory for the plugins. ') + hintMessage;
```
Comment (msujew) [inline packages/plugin-ext/src/hosted/node/hosted-plugin-process.ts:224]: Minor: Whitespace can sometimes get lost in automatic translation.
```suggestion
            hintMessage = `${nls.localize('theia/plugin-ext/runtimeMemoryError', 'Probably there is not enough memory for the plugins.')} {hintMessage}`;
```

**Review by msujew — CHANGES_REQUESTED** [review]
Looking over this, I've noticed that this code actually resides in the backend part of the application. While `nls.localize` is callable from backend code, it will always return the english default value. This approach to fixing this issue isn't working.

In theory, you will need a dedicated frontend service that performs the `MessageService#error` call with a localized value.

**Comment by Zebsterpasha** [comment]
@msujew This problem is not critical for me, I will not be able to fix it in the near future.

**Comment by msujew** [comment]
Alright, let's close it for now. Maybe someone else will take inspiration from this to fix the issue.

**Pull request closed without merging**